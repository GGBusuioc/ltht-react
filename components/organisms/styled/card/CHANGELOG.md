# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.2](https://ssh.github.com/ltht-epr/ltht-react/compare/@ltht-react/card@0.1.1...@ltht-react/card@0.1.2) (2020-02-27)

**Note:** Version bump only for package @ltht-react/card





## [0.1.1](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/card@0.1.0...@ltht-react/card@0.1.1) (2020-02-26)

**Note:** Version bump only for package @ltht-react/card
