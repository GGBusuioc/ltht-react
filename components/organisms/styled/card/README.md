
# Card

<!-- STORY -->

### Import

```js
import Card from '@ltht-react/card'
```

### Usage

```jsx
<Card />
```
  