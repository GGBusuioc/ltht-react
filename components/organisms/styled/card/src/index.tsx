export * from './organisms'
export * from './store'
