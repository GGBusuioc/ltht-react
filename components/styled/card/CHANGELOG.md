# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.8](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/card@0.1.7...@ltht-react/card@0.1.8) (2020-04-06)

**Note:** Version bump only for package @ltht-react/card






## [0.1.7](https://ssh.github.com/ltht-epr/ltht-react/compare/@ltht-react/card@0.1.6...@ltht-react/card@0.1.7) (2020-04-06)

**Note:** Version bump only for package @ltht-react/card





## [0.1.6](https://ssh.github.com/ltht-epr/ltht-react/compare/@ltht-react/card@0.1.5...@ltht-react/card@0.1.6) (2020-04-03)

**Note:** Version bump only for package @ltht-react/card





## [0.1.5](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/card@0.1.4...@ltht-react/card@0.1.5) (2020-03-16)

**Note:** Version bump only for package @ltht-react/card






## [0.1.4](https://ssh.github.com/ltht-epr/ltht-react/compare/@ltht-react/card@0.1.3...@ltht-react/card@0.1.4) (2020-03-12)

**Note:** Version bump only for package @ltht-react/card





## [0.1.3](https://ssh.github.com/ltht-epr/ltht-react/compare/@ltht-react/card@0.1.2...@ltht-react/card@0.1.3) (2020-03-06)

**Note:** Version bump only for package @ltht-react/card





## [0.1.2](https://ssh.github.com/ltht-epr/ltht-react/compare/@ltht-react/card@0.1.1...@ltht-react/card@0.1.2) (2020-02-27)

**Note:** Version bump only for package @ltht-react/card





## [0.1.1](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/card@0.1.0...@ltht-react/card@0.1.1) (2020-02-26)

**Note:** Version bump only for package @ltht-react/card
