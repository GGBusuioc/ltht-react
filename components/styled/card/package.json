{
  "name": "@ltht-react/card",
  "version": "0.1.10",
  "description": "ltht-react styled Card component.",
  "author": "LTHT",
  "homepage": "",
  "license": "MIT",
  "main": "lib/index.js",
  "typings": "lib/index.d.ts",
  "files": [
    "lib"
  ],
  "directories": {
    "lib": "lib"
  },
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "clean": "rimraf lib",
    "build": "yarn build:lib",
    "watch": "tsc -w",
    "build:lib": "tsc",
    "type:check": "tsc --noEmit",
    "lint": "prettylint 'src/*.{ts,tsx}'"
  },
  "dependencies": {
    "@emotion/core": "^10.0.27",
    "@emotion/styled": "^10.0.27",
    "@ltht-react/icon": "^0.1.7",
    "@ltht-react/styles": "^0.1.3",
    "@ltht-react/types": "^0.0.12",
    "@ltht-react/utils": "^0.1.5",
    "emotion": "^10.0.27",
    "react": "^16.12.0"
  }
}
