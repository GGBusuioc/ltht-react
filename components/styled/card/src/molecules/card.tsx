/** @jsx jsx */
import React from 'react'
import { css, jsx, SerializedStyles } from '@emotion/core'

import { CSS_RESET, TEXT_COLOURS, CARD_BACKGROUND_COLOUR, CARD_BACKGROUND_COLOUR_NO_DATA } from '@ltht-react/styles'
// import { CardProvider, ProviderProps } from '../store/card-context'
import CardEmpty from '../atoms/card-empty'

const styles = (noData: boolean): SerializedStyles => {
  return css`
  ${CSS_RESET}
  background: ${noData ? CARD_BACKGROUND_COLOUR_NO_DATA : CARD_BACKGROUND_COLOUR};
  color: ${noData ? TEXT_COLOURS.SECONDARY : TEXT_COLOURS.PRIMARY};
  margin-bottom: 0.5rem;
  border-radius: 4px;
  box-shadow: 
    0px 2px 1px -1px rgba(102, 102, 102, 0.1), 
    0px 1px 1px 0px rgba(102, 102, 102, 0.15), 
    0px 1px 3px 0px rgba(102, 102, 102,.6);
  margin: 0 5px 10px 0;
  -webkit-font-smoothing: antialiased;
`
}

const Card: React.FC<Props> = ({ children, noData = false }) => {
  return (
    // <CardProvider
    //   collapsible={collapsible}
    //   collapsed={collapsed}
    //   noData={noData}
    //   childrenCount={React.Children.count(children)}
    // >
    <div css={styles(noData)}>
      {children}
      {noData && <CardEmpty />}
    </div>
    // </CardProvider>
  )
}

interface Props {
  noData?: boolean
}

export default Card
