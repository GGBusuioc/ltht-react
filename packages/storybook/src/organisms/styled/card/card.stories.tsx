import React from 'react'
import { storiesOf } from '@storybook/react'
import JSXAddon from 'storybook-addon-jsx'

import readme from '@ltht-react/card/README.md'
import Card from '@ltht-react/card'
import AllergySummary from '@ltht-react/allergy-summary'
import { InfoSummary, WarningSummary, ErrorSummary, MissingDataSummary } from '@ltht-react/summary'
import allergies from '../../clinical/allergies/allergy.fixtures'

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const stories = storiesOf('Organisms - Styled|Card', module) as any

stories.addWithJSX = JSXAddon.addWithJSX

stories.addParameters({
  readme: {
    sidebar: readme,
  },
})

const clickHandler = (): void => {
  // eslint-disable-next-line no-console
  console.log('Handler clicked')
}

stories.addWithJSX('Basic', () => (
  <>
    <Card>
      <Card.Header>
        <Card.Title>Header</Card.Title>
      </Card.Header>
      <Card.Body>
        <Card.Subtitle>Subtitle</Card.Subtitle>
        <Card.Text>Body Text</Card.Text>
      </Card.Body>
    </Card>
  </>
))

stories.addWithJSX('With Footer', () => (
  <Card>
    <Card.Header>
      <Card.Title>Header</Card.Title>
    </Card.Header>
    <Card.Body>
      <Card.Subtitle>Subtitle</Card.Subtitle>
      <Card.Text>Body Text</Card.Text>
    </Card.Body>
    <Card.Footer>Footer</Card.Footer>
  </Card>
))

stories.addWithJSX('With List', () => (
  <div>
    <Card>
      <Card.Header>
        <Card.Title>Header</Card.Title>
      </Card.Header>
      <Card.List>
        <Card.ListItem>Item 1</Card.ListItem>
        <Card.ListItem>Item 2</Card.ListItem>
      </Card.List>
    </Card>
  </div>
))

stories.addWithJSX('List (Clickable)', () => (
  <div>
    <Card>
      <Card.Header>
        <Card.Title>Header</Card.Title>
      </Card.Header>
      <Card.List>
        <Card.ListItem onClick={clickHandler}>
          <div>Item 1</div>
        </Card.ListItem>
        <Card.ListItem onClick={clickHandler}>
          <div>Item 2</div>
        </Card.ListItem>
      </Card.List>
    </Card>
  </div>
))

stories.addWithJSX('Info Summary', () => (
  <div>
    <Card>
      <Card.Header>
        <Card.Title>Allergies</Card.Title>
      </Card.Header>
      <InfoSummary clickHandler={clickHandler} />
      <Card.List>
        {allergies.map(allergy => (
          <Card.ListItem>
            <AllergySummary allergy={allergy} />
          </Card.ListItem>
        ))}
      </Card.List>
      <InfoSummary clickHandler={clickHandler} />
      <Card.List>
        <Card.ListItem>Missing 10</Card.ListItem>
      </Card.List>
      <WarningSummary clickHandler={clickHandler} />
      <Card.List>
        <Card.ListItem>Missing 10</Card.ListItem>
        <Card.ListItem>Missing 10</Card.ListItem>
      </Card.List>
    </Card>
  </div>
))

stories.addWithJSX('Warning Summary', () => (
  <div>
    <Card>
      <Card.Header>
        <Card.Title>Allergies</Card.Title>
      </Card.Header>
      <WarningSummary clickHandler={clickHandler} />
      <Card.List>
        {allergies.map(allergy => (
          <Card.ListItem>
            <AllergySummary allergy={allergy} />
          </Card.ListItem>
        ))}
      </Card.List>
    </Card>
  </div>
))

stories.addWithJSX('Error Summary', () => (
  <div>
    <Card>
      <Card.Header>
        <Card.Title>Allergies</Card.Title>
      </Card.Header>
      <ErrorSummary clickHandler={clickHandler} />
      <Card.List>
        <Card.ListItem>No Data</Card.ListItem>
      </Card.List>
    </Card>
  </div>
))

stories.addWithJSX('Missing Data Summary', () => (
  <Card>
    <Card.Header>
      <Card.Title>Allergies</Card.Title>
    </Card.Header>
    <MissingDataSummary
      noData={['Item 1', 'Item 2']}
      missingData={['Item 1', 'Item 2']}
      partialData={['Item 1', 'Item 2']}
      completeData={['Item 1', 'Item 2']}
    />
  </Card>
))

stories.addWithJSX('No Data', () => (
  <div>
    <Card>
      <Card.Header>
        <Card.Title>Allergies</Card.Title>
      </Card.Header>
    </Card>
  </div>
))
