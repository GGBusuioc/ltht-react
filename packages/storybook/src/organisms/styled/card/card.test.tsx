import React from 'react'
import { mount } from 'enzyme'
import ReactDOM from 'react-dom'

import { Card, CardBody, CardHeader, CardFooter, CardList, CardListItem } from '@ltht-react/card'

const components = (
  <>
    <Card>
      <CardHeader>
        <h3>Heading</h3>
      </CardHeader>
      <CardBody>Body</CardBody>
      <CardFooter>Footer</CardFooter>
    </Card>
    <Card>
      <CardHeader>
        <h3>Heading</h3>
      </CardHeader>
      <CardList>
        <CardListItem>Item 1</CardListItem>
        <CardListItem>Item 2</CardListItem>
      </CardList>
      <CardFooter>Footer</CardFooter>
    </Card>
  </>
)

describe('Card', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(components, div)
  })
  it('matches snapshot', () => {
    expect(mount(components)).toMatchSnapshot('wrapper mount')
  })
})
