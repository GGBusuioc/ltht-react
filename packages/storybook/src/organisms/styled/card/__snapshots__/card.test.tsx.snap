// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches snapshot: wrapper mount 1`] = `
Array [
  .emotion-3 {
  color: #231f20;
  font-family: 'SF Pro Text','SF Pro Icons','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.8rem;
  background: #FFFFFF;
  color: #231f20;
  margin-bottom: 0.5rem;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(102,102,102,0.1), 0px 1px 1px 0px rgba(102,102,102,0.15), 0px 1px 3px 0px rgba(102,102,102,.6);
  margin: 0 5px 10px 0;
  -webkit-font-smoothing: antialiased;
}

.emotion-3 *,
.emotion-3 ::before,
.emotion-3 ::after {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
}

.emotion-3 * {
  margin: 0;
  padding: 0;
}

.emotion-3 h3 {
  font-size: 1rem;
}

@media only screen and (min-width:768px) {
  .emotion-3 {
    font-size: 0.9rem;
  }

  .emotion-3 h3 {
    font-size: 1.1rem;
  }
}

@media only screen and (min-width:1024px) {
  .emotion-3 {
    font-size: 1rem;
  }

  .emotion-3 h3 {
    font-size: 1.2rem;
  }
}

@media only screen and (min-width:1200px) {
  .emotion-3 {
    font-size: 1.1rem;
  }

  .emotion-3 h3 {
    font-size: 1.3rem;
  }
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0.5rem;
}

.emotion-1 {
  padding: 0.5rem;
}

.emotion-1:before {
  content: '';
  display: block;
  border-top: 1px solid #b0b0b0;
  margin: -0.5rem -0.5rem 0.5rem 0;
}

.emotion-2 {
  padding: 0.5rem;
}

<Card>
    <div
      className="emotion-3"
    >
      <CardHeader>
        <div
          className="emotion-0"
          onClick={[Function]}
          title=""
        >
          <div>
            <h3>
              Heading
            </h3>
          </div>
        </div>
      </CardHeader>
      <CardBody>
        <div
          className="emotion-1"
        >
          Body
        </div>
      </CardBody>
      <CardFooter>
        <div
          className="emotion-2"
        >
          Footer
        </div>
      </CardFooter>
    </div>
  </Card>,
  .emotion-3 {
  color: #231f20;
  font-family: 'SF Pro Text','SF Pro Icons','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.8rem;
  background: #FFFFFF;
  color: #231f20;
  margin-bottom: 0.5rem;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(102,102,102,0.1), 0px 1px 1px 0px rgba(102,102,102,0.15), 0px 1px 3px 0px rgba(102,102,102,.6);
  margin: 0 5px 10px 0;
  -webkit-font-smoothing: antialiased;
}

.emotion-3 *,
.emotion-3 ::before,
.emotion-3 ::after {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
}

.emotion-3 * {
  margin: 0;
  padding: 0;
}

.emotion-3 h3 {
  font-size: 1rem;
}

@media only screen and (min-width:768px) {
  .emotion-3 {
    font-size: 0.9rem;
  }

  .emotion-3 h3 {
    font-size: 1.1rem;
  }
}

@media only screen and (min-width:1024px) {
  .emotion-3 {
    font-size: 1rem;
  }

  .emotion-3 h3 {
    font-size: 1.2rem;
  }
}

@media only screen and (min-width:1200px) {
  .emotion-3 {
    font-size: 1.1rem;
  }

  .emotion-3 h3 {
    font-size: 1.3rem;
  }
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0.5rem;
}

.emotion-2 {
  padding: 0.5rem;
}

.emotion-1 {
  list-style: none;
  padding-left: 0.5rem;
}

.emotion-1 li {
  border-top: 1px solid #b0b0b0;
  padding: 0.5rem 0.5rem 0.5rem 0;
}

<Card>
    <div
      className="emotion-3"
    >
      <CardHeader>
        <div
          className="emotion-0"
          onClick={[Function]}
          title=""
        >
          <div>
            <h3>
              Heading
            </h3>
          </div>
        </div>
      </CardHeader>
      <CardList>
        <ul
          className="emotion-1"
        >
          <ListItem>
            <li>
              Item 1
            </li>
          </ListItem>
          <ListItem>
            <li>
              Item 2
            </li>
          </ListItem>
        </ul>
      </CardList>
      <CardFooter>
        <div
          className="emotion-2"
        >
          Footer
        </div>
      </CardFooter>
    </div>
  </Card>,
]
`;
